core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -641,21 +643,30 @@ private ServiceRequestContext newEarlyRespondingRequestContext(Channel channel,
         return new DefaultServiceRequestContext(
                 serviceConfig,
                 channel, NoopMeterRegistry.get(), protocol(),
-                nextRequestId(), routingCtx, routingResult, req.exchangeType(),
+                nextRequestId(routingCtx), routingCtx, routingResult, req.exchangeType(),
                 req, sslSession, proxiedAddresses, clientAddress,
                 System.nanoTime(), SystemInfo.currentTimeMicros());
     }
 
-    private RequestId nextRequestId() {
-        final RequestId id = config.requestIdGenerator().get();
-        if (id == null) {
-            if (!warnedNullRequestId) {
-                warnedNullRequestId = true;
-                logger.warn("requestIdGenerator.get() returned null; using RequestId.random()");
+    private RequestId nextRequestId(RoutingContext routingCtx) {
+        try {
+            final RequestId id = config.requestIdGenerator().apply(routingCtx);
+            if (id == null) {
+                if (!warnedNullRequestId) {
+                    warnedNullRequestId = true;
+                    logger.warn("requestIdGenerator.apply(routingCtx) returned null; using RequestId.random()");
+                }
+                return RequestId.random();
+            } else {
+                return id;
+            }
```

Comment:
The core logic of this method is to return a normal `RequestId`. The nested if-else and the error handling hide it. What do you think of using early-return to locate the main login first and others behind?
```suggestion
            final RequestId id = config.requestIdGenerator().apply(routingCtx);
            if (id != null) {
                return id;
            }
            
            if (!warnedNullRequestId) {
                warnedNullRequestId = true;
                logger.warn("requestIdGenerator.apply(routingCtx) returned null; using RequestId.random()");
            }
            return RequestId.random();
```

Reply:
Thank you for suggestion! I apply it.